src/Symfony/Bundle/SecurityBundle/Templating/Helper/LogoutUrlHelper.php

```
@@ -25,20 +27,32 @@
  */
 class LogoutUrlHelper extends Helper
 {
-    private $container;
+    private $requestStack;
     private $listeners = array();
     private $router;
+    private $tokenStorage;
 
     /**
      * Constructor.
      *
-     * @param ContainerInterface    $container A ContainerInterface instance
-     * @param UrlGeneratorInterface $router    A Router instance
+     * @param ContainerInterface|RequestStack $container    A ContainerInterface instance or RequestStack
+     * @param UrlGeneratorInterface           $router       The router service
+     * @param TokenStorageInterface|null      $tokenStorage The token storage service
+     *
+     * @deprecated Passing a ContainerInterface as a first argument is deprecated since 2.7 and will be removed in 3.0.
      */
-    public function __construct(ContainerInterface $container, UrlGeneratorInterface $router)
+    public function __construct($container, UrlGeneratorInterface $router, TokenStorageInterface $tokenStorage = null)
     {
-        $this->container = $container;
+        if ($container instanceof ContainerInterface) {
+            $this->requestStack = $container->get('request_stack');
+        } elseif ($container instanceof RequestStack) {
+            $this->requestStack = $container;
+        } else {
+            throw new \BadMethodCallException(sprintf('%s takes either a RequestStack or a ContainerInterface object as its first argument.', __METHOD__));
```

please throw \InvalidArgumentException instead